src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3396,18 +3393,16 @@ void DoorLockServer::SendLockOperationEvent(chip::EndpointId endpointId, LockOpe
 
 void DoorLockServer::ScheduleAutoRelock(chip::EndpointId endpointId, uint32_t timeoutSec)
 {
-    emberEventControlSetInactive(&AutolockEvent);
-
-    AutolockEvent.endpoint = endpointId;
-    AutolockEvent.callback = DoorLockOnAutoRelockCallback;
-
     uint32_t timeoutMs =
         (DOOR_LOCK_MAX_LOCK_TIMEOUT_SEC >= timeoutSec) ? timeoutSec * MILLISECOND_TICKS_PER_SECOND : DOOR_LOCK_MAX_LOCK_TIMEOUT_SEC;
-    auto err = emberEventControlSetDelayMS(&AutolockEvent, timeoutMs);
 
-    if (EMBER_SUCCESS != err)
+    CHIP_ERROR err =
+        DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(timeoutMs), DoorLockOnAutoRelockCallback,
+                                              reinterpret_cast<void *>(static_cast<uintptr_t>(endpointId)));
+
+    if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(Zcl, "Failed to schedule autorelock: timeout=%" PRIu32 ", status=0x%x", timeoutSec, err);
+        ChipLogError(Zcl, "Failed to schedule autorelock: timeout=%" PRIu32 ", status=0x%" PRIx32, timeoutSec, err.AsInteger());
```

Comment:
This still has the AsInteger thing?  As in, the accepted suggestion got overridden by a force-push or something?

Reply:
And note the bit about the workflow file above.